streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/promise/Promise.kt

```diff
@@ -89,11 +89,16 @@ class Promise<T> internal constructor(
         onError = apply
     )
 
-    fun finally(execute: () -> Unit): Publisher<T> = thenReturn(
+    fun finally(execute: () -> Unit): Promise<T> = thenReturn(
```

Comment:
Maybe change the param name to onFinally ?

Reply:
I tried to keep the naming consistent with the other methods, such as `onSuccess(accept: ...)` and `onErrorReturn(apply: ...)`, where : 

- **accept** is a consumer block`accept: (value) -> Unit`
- **apply** is a transformer block `apply: (value) -> Promise`
- **execute** is a simple block `execute: () -> Unit`
- **supply** is a supplier block `supply: () -> Promise`

Using finally(onFinally: ..) seemed repetitive, just like `onSuccess(onSuccess: ...)`.

I can change them everywhere, though, if we feel like it's more familiar.

Reply:
But you also have `then` with `onSuccess` and `onError` which is also not consistent. 

Also, on another subject, I'm curious on why we chose to use onSuccess and onError rather than `onFullfilled` and `onRejected` ?

Reply:
> But you also have then with onSuccess and onError which is also not consistent.

Right. I'll leave this as is for now, and if we feel like it feels weird along the way, we could easily change it.

Reply:
> Also, on another subject, I'm curious on why we chose to use onSuccess and onError rather than onFullfilled and onRejected ?

The `success` and `error` naming felt more familiar when implementing the Promises, since these keywords are often used in our Reactive world at Mirego. I kind of like the `fullfiled` and `rejected` naming as they are really bound to a Promise state, but I felt like using familiar word would be easier for people to adopt the interfaces. Again, this is still something we can change along the way :)